spswarehouse/powerschool/powerschool_calpads.py

```diff
@@ -77,101 +67,28 @@ def download_calpads_report_for_school(self, school_full_name: str, submission_w
-                    file_postfix=file_postfix, 
-                    destination_directory_path=destination_directory_path, 
-                    report_parameters=report_parameters,
-                    validation_only_run=validation_only_run
-                    )
-            elif(calpads_report_abbreviation == 'SCSC'):
-                return self._download_eoy_report_for_student_course_section_records_scsc(
-                    ps_report_link_text=PS_REPORT_LINK_TEXT[calpads_report_abbreviation],
-                    file_postfix=file_postfix, 
-                    destination_directory_path=destination_directory_path, 
-                    report_parameters=report_parameters,
-                    # The below is an additional parameter compared to the function calls earlier in the 
-                    #   if-else tree
-                    ps_school_subdistrict_name=ps_school_subdistrict_name, 
-                    validation_only_run=validation_only_run,
-                    )
-            else:
-                raise Exception("CALPADS EOY report name not supported")
-        elif(submission_window == 'All Year'):
-            if(calpads_report_abbreviation == 'SENR'):
-                return self._download_all_year_report_for_ssid_enrollment_records_senr(
-                    ps_report_link_text=PS_REPORT_LINK_TEXT[calpads_report_abbreviation],
-                    file_postfix=file_postfix, 
-                    destination_directory_path=destination_directory_path, 
-                    report_parameters=report_parameters,
-                    validation_only_run=validation_only_run,
-                    )
-            elif(calpads_report_abbreviation == 'SELA'):
-                return self._download_all_year_report_for_student_english_language_acquisition_records_sela(
-                    ps_report_link_text=PS_REPORT_LINK_TEXT[calpads_report_abbreviation],
-                    file_postfix=file_postfix, 
-                    destination_directory_path=destination_directory_path, 
-                    report_parameters=report_parameters,
-                    validation_only_run=validation_only_run,
-                    )
-            elif(calpads_report_abbreviation == 'SINF'):
-                return self._download_all_year_report_for_student_information_records_sinf(
-                    ps_report_link_text=PS_REPORT_LINK_TEXT[calpads_report_abbreviation],
-                    file_postfix=file_postfix, 
-                    destination_directory_path=destination_directory_path, 
-                    report_parameters=report_parameters,
-                    validation_only_run=validation_only_run,
-                    )
-            else:
-                raise Exception("CALPADS All Year report name not supported")
-        else:
-            raise Exception("Submission window name not supported")
-        
+        return self.CALPADS_REPORT_TYPES[calpads_report_abbreviation]['function'](self, **report_kwargs)
```

Comment:
Looking good! The `(self` is a little weird - you might want to add a comment to explain why it's needed

Other than that, I'd recommend first adding an explicit check for `calpads_report_abbreviation in self.CALPADS_REPORT_TYPES` so that we can raise a more helpful exception than "KeyError"